stable_baselines3/common/policies.py

```diff
@@ -260,14 +275,29 @@ def predict(
         #     state = self.initial_state
         # if mask is None:
         #     mask = [False for _ in range(self.n_envs)]
-        if isinstance(observation, dict):
+        # Need to check the observation if its a ObsDictWrapper
+
+        if isinstance(self.observation_space, ObsDictWrapper):
             observation = ObsDictWrapper.convert_dict(observation)
-        else:
-            observation = np.array(observation)
+        elif isinstance(observation, dict):
+            # need to copy the dict as the dict in VecFrameStack will become a torch tensor
+            observation = copy.deepcopy(observation)
+            for key, obs in observation.items():
+                obs_space = self.observation_space.spaces[key]
+                if is_image_space(obs_space):
+                    obs = np.array(obs)
+                    if not (obs.shape == obs_space.shape or obs.shape[1:] == obs_space.shape):
```

Comment:
looks like duplicated code from above... maybe a separate (private) helper would be better?

Reply:
True, I created `try_transpose_img_observation(observation, observation_space)` on `BasePolicy` but we can move it to a util script later.